itest/integration_test.go

```diff
@@ -1494,8 +1495,27 @@ func TestPauseAndResume(t *testing.T) {
 			sv := testutil.NewStubbedValidator()
 			sv.StubResult(datatransfer.ValidationResult{Accepted: true})
 			sv.StubRestartResult(datatransfer.ValidationResult{Accepted: true})
-
 			var chid datatransfer.ChannelID
+
+			gsData.Gs1.RegisterOutgoingBlockHook(func(p peer.ID, r graphsync.RequestData, block graphsync.BlockData, ha graphsync.OutgoingBlockHookActions) {
+				if block.Index() == 5 && block.BlockSizeOnWire() > 0 {
```

Comment:
so the flake in here was a race on where the pause was triggered? and shifting further down the pipe to the hook, with added delay, fixed it?